src/test/java/seedu/address/logic/commands/edit/EditInternshipCommandTest.java

```diff
@@ -1,2 +1,182 @@
+import static seedu.address.testutil.TypicalIndexes.INDEX_FIRST_ITEM;
+import static seedu.address.testutil.TypicalIndexes.INDEX_THIRD_ITEM;
+import static seedu.address.testutil.TypicalInternship.GOOGLE;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.commands.results.CommandResult;
+import seedu.address.model.item.Internship;
+import seedu.address.testutil.InternshipBuilder;
+import seedu.address.testutil.ModelStub;
+
+public class EditInternshipCommandTest {
+
+    private Internship sampleEditedInternship = GOOGLE;
+
+    private EditInternshipDescriptor editInternshipDescriptor = new EditInternshipDescriptor();
+
+    @Test
+    public void constructor_nullInternship_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new EditInternshipCommand(null, null));
+    }
+
+    @Test
+    public void execute_invalidIndex_throwsCommandException() {
+        Internship validItem = new InternshipBuilder().build();
+        ModelStub modelStub = new EditInternshipCommandTest.ModelStubWithInternship(validItem);
+        Index invalidIndex = INDEX_THIRD_ITEM;
+        setEditInternshipDescriptor();
+        EditInternshipCommand editInternshipCommand = new EditInternshipCommand(invalidIndex, editInternshipDescriptor);
+        assertThrows(CommandException.class,
+                Messages.MESSAGE_INVALID_INDEX, () -> editInternshipCommand.execute(modelStub));
+    }
+
+    @Test
+    public void execute_validIndex_editSuccessful() throws CommandException {
+        ModelStubContainingInternshipEdited modelStub = new ModelStubContainingInternshipEdited(sampleEditedInternship);
+        setEditInternshipDescriptor();
+        Index validIndex = INDEX_FIRST_ITEM;
+
+        Internship toEdit = modelStub.getInternshipByIndex(validIndex);
+        EditInternshipCommand editInternshipCommand = new EditInternshipCommand(validIndex, editInternshipDescriptor);
+        assertEquals(Arrays.asList(toEdit), modelStub.internships);
+
+        CommandResult commandResult = editInternshipCommand.execute(modelStub);
+
+        assertEquals(String.format(EditInternshipCommand.MESSAGE_EDIT_INTERNSHIP_SUCCESS, sampleEditedInternship),
+                commandResult.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_duplicateInternship_throwsCommandException() throws CommandException {
+        Index validIndex = INDEX_FIRST_ITEM;
+        setEditInternshipDescriptor();
+        EditInternshipCommand editCommand = new EditInternshipCommand(validIndex, editInternshipDescriptor);
+        ModelStubContainingInternshipEdited modelStub = new ModelStubContainingInternshipEdited(
+                new InternshipBuilder().build());
+        modelStub.addInternship(sampleEditedInternship);
```

Comment:
This part i tried assertThrows exception but it does not throw and the modelStub just edit the first Internship into a duplicate of the second Internship

Reply:
And I don't include this @Test in Skill, Project and Resume yet! Just put here as a consideration